py/dml/c_backend.py

```diff
@@ -2853,6 +2884,9 @@ def generate_cfile_body(device, footers, full_module, filename_prefix):
         register_saved_attributes(init_code, device)
         generate_init(device, init_code, filename_prefix)
 
+    if serialize.serialized_traits:
```

Comment:
`if` can be removed, one empty function is not a cost.

Reply:
The time it takes to dig through all nodes of the device pointlessly _is_ .

Reply:
I doubt it! No need to bother about a small O(n). If you insist, maybe move the micro-optimization to around that loop?

(nitpick, does not block approval)

Reply:
But yes, I can move the optimization.